### PR TITLE
introducing a new section in standards - release notes

### PR DESCRIPTION
**Pull request by pritidesai**
Adding details on how to write a release note. Also provided a couple of samples of good and bad release notes.

### CODE REVIEW

**Review by dibyom — APPROVED** [review]

**Review by vdemeester — APPROVED** [review]

**Comment by pritidesai** [comment]
thank you @dibyom and @vdemeester for the review 🙏 

Once this is merged, I will update the pipeline PR template (can be extended to other projects) to have a reference to this section.

@tektoncd/core-maintainers please take a look!

**Comment by dibyom** [comment]
> Once this is merged, I will update the pipeline PR template (can be extended to other projects) to have a reference to this section.

Perfect!

**Review by jerop — APPROVED** [review]

**Pull request merged**
